src/components/cover-image.tsx

```diff
@@ -1,34 +1,70 @@
 import cn from "classnames";
 import Link from "next/link";
+import Image from "next/image";
 import { postRoute } from "../routes";
 
-type Props = {
-  title: string;
+type PostHeaderImageProps = {
   src: string;
-  slug?: string;
+  alt: string;
+  width: number;
+  height: number;
 };
 
-const CoverImage = ({ title, src, slug }: Props) => {
-  const image = (
-    <img
-      src={src}
-      alt={`Cover Image for ${title}`}
-      className={cn("shadow-small", {
-        "hover:shadow-medium transition-shadow duration-200": slug,
-      })}
-    />
+export const PostHeaderImage = ({
+  src,
+  alt,
+  width,
+  height,
+}: PostHeaderImageProps) => {
+  return (
+    <div className="sm:mx-0">
+      <div className="relative w-full">
+        <Image
+          src={src}
+          alt={alt}
+          layout="responsive"
+          width={width}
+          height={height}
+        />
+      </div>
+    </div>
   );
+};
+
+type PreviewImageProps = {
+  title: string;
+  src: string;
+  alt: string;
+  width: number;
+  height: number;
+  slug: string;
+  heroPostImage?: boolean;
```

Comment:
A suggestion to make more explicit why each props is needed:

```ts
type PreviewImageProps = {
  targetTitle: string;
  targetUrl: string;
  
  imageSrc: string;
  imageAlt: string;
  imageWidth: number;
  imageHeight: number;
  
  hero?: boolean;
}
```

Reply:
```
  blogPostTitle: string;
  blogPostUrl: string;
  imageSrc: string;
  imageAlt: string;
  imageWidth: number;
  imageHeight: number;
  heroPostImage?: boolean;
```